connmgr/seed.go

```diff
@@ -0,0 +1,62 @@
+package connmgr
+
+import (
+	mrand "math/rand"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/wire"
+)
+
+const (
+	// These constants are used by the DNS seed code to pick a random last
+	// seen time.
+	secondsIn3Days int32 = 24 * 60 * 60 * 3
+	secondsIn4Days int32 = 24 * 60 * 60 * 4
+)
+
+// onSeed is the signature of the callback function which is invoked when DNS
+// seeding is succesfull.
+type onSeed func(addrs []*wire.NetAddress)
+
+// lookupFunc is the signature of the DNS lookup function.
+type lookupFunc func(string) ([]net.IP, error)
+
+// SeedFromDNS uses DNS seeding to populate the address manager with peers.
+func SeedFromDNS(chainParams *chaincfg.Params, lookupFn lookupFunc, seedFn onSeed) {
```

Comment:
Using unexported types in functions is generally not a good thing to do.  I'd suggest exporting the `OnSeed` and `LookupFunc` types.